api/lib/validation/attributeValidator.js

```diff
@@ -0,0 +1,230 @@
+'use strict'
+
+var moment = require('moment');
+require('moment-range');
+var intersect = require('intersect');
+var logger = require('../log/logger');
+
+var createErrorResponse = function(property,message,instance,stack) {
+  var errorObject = { property:property,
+  message: message,instance:instance,stack:stack }; 
+  return errorObject;
+}
+
+var compareTimes = function(startTime,endTime) {
+  var result = true;
+  if(moment(startTime,'HH:mm').isAfter(moment(endTime,'HH:mm'))) {
+    result = false;
+  }
+  return result;
+}
+
+var compareDateTimes = function(startDateTime,endDateTime) {
+  var result = true;
+  if(moment(startDateTime,'YYYY-MM-DDTHH:mm')
+    .isAfter(moment(endDateTime,'YYYY-MM-DDTHH:mm'))) {
+    result = false;
+  }
+  return result;
+}
+
+var validateDaysOfWeekErrors = function(inputRecurringSchedules) {
+  var daysOfWeekErrors = [];
+  var errorCount = 0;
+  for(let j = 0; j < inputRecurringSchedules.length; j++) {
+    for(let i = j + 1; i < inputRecurringSchedules.length; i++) {
+      if(inputRecurringSchedules[i].hasOwnProperty('days_of_week') && 
+        inputRecurringSchedules[j].hasOwnProperty('days_of_week')) {
+        if(intersect(inputRecurringSchedules[i].days_of_week,
+         inputRecurringSchedules[j].days_of_week).length > 0) {
+          if(inputRecurringSchedules[i].start_time <=
+          inputRecurringSchedules[j].end_time && 
+          inputRecurringSchedules[i].end_time >=
+          inputRecurringSchedules[j].start_time) {
+            let error = createErrorResponse('recurring_schedule.start_time',
+            'recurring_schedule.start_time and recurring_schedule.end_time ' + 
+            'ranges are overlapping',inputRecurringSchedules[j],
+            'Days_of_week based time range of recurring_schedule[' + j + ']' + 
```

Comment:
That's a good point. We also wanted to get your feedback on the error response that we create. An example of a validation error is this (in case the metric_type is misspelt). The response is loosely based on the response created by the schema validator that we use https://www.npmjs.com/package/jsonschema
```
{
    "success": false,
    "error": [
        {
            "property": "instance.scaling_rules[0].metric_type",
            "message": "is not one of enum values: MemoryUsage",
            "schema": {
                "type": "string",
                "enum": [
                    "MemoryUsage"
                ]
            },
            "instance": "MemoryUsagde",
            "name": "enum",
            "argument": [
                "MemoryUsage"
            ],
            "stack": "instance.scaling_rules[0].metric_type is not one of enum values: MemoryUsage"
        }
    ],
    "result": null
}
```